src/cursor/cur_hs.c

```diff
@@ -658,6 +658,87 @@ __curhs_insert(WT_CURSOR *cursor)
     API_END_RET(session, ret);
 }
 
+/*
+ * __curhs_update --
+ *     WT_CURSOR->update method for the hs cursor type.
+ */
+static int
+__curhs_update(WT_CURSOR *cursor)
+{
+    WT_CURSOR *file_cursor;
+    WT_CURSOR_BTREE *cbt;
+    WT_CURSOR_HS *hs_cursor;
+    WT_DECL_ITEM(hs_value);
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+    WT_UPDATE *hs_tombstone, *hs_upd;
+
+    uint64_t hs_upd_type;
+    wt_timestamp_t hs_durable_ts, hs_stop_durable_ts;
+
+    hs_cursor = (WT_CURSOR_HS *)cursor;
+    file_cursor = hs_cursor->file_cursor;
+    cbt = (WT_CURSOR_BTREE *)file_cursor;
+    hs_tombstone = hs_upd = NULL;
+
+    CURSOR_API_CALL_PREPARE_ALLOWED(cursor, session, update, CUR2BT(file_cursor));
+
+    /* We are assuming that the caller has already searched and found the key. */
+    WT_ASSERT(session, F_ISSET(file_cursor, WT_CURSTD_KEY_SET | WT_CURSTD_VALUE_SET));
```

Comment:
I mean we should assert F_ISSET(hs_cursor, WT_HS_CUR_COUNTER_SET) also assert WT_ASSERT(session, F_ISSET(file_cursor, WT_CURSTD_KEY_INT)).

Reply:
I am asserting on HS cursor flags further down. I'll add a check for  `WT_CURSTD_KEY_INT` here.